pkg/kubectl/resource/builder.go

```diff
@@ -598,10 +598,27 @@ func (b *Builder) SingleResourceType() *Builder {
 	return b
 }
 
-// mappingFor returns the RESTMapping for the Kind referenced by the resource.
-// prefers a fully specified GroupVersionResource match.  If we don't have one match on GroupResource
-func (b *Builder) mappingFor(resourceArg string) (*meta.RESTMapping, error) {
-	fullySpecifiedGVR, groupResource := schema.ParseResourceArg(resourceArg)
+// mappingFor returns the RESTMapping for the Kind given, or the Kind referenced by the resource.
+// prefers a fully specified GroupVersionKind match. If we don't have one, match on a fully specified
+// GroupVersionResource, or fallback to a match on GroupResource.
+func (b *Builder) mappingFor(resourceOrKindArg string) (*meta.RESTMapping, error) {
+	fullySpecifiedGVK, groupKind := schema.ParseKindArg(resourceOrKindArg)
+	if fullySpecifiedGVK == nil {
+		gvk := groupKind.WithVersion("")
+		fullySpecifiedGVK = &gvk
+	}
+
+	if !fullySpecifiedGVK.Empty() {
+		if mapping, err := b.mapper.RESTMapping(fullySpecifiedGVK.GroupKind(), fullySpecifiedGVK.Version); err == nil {
+			return mapping, nil
+		} else {
+			if mapping, err := b.mapper.RESTMapping(groupKind, ""); err == nil {
```

Comment:
@deads2k thanks, comments addressed. However both `b.mapper.RESTMapping` here will return a "not matched" error if only the kind is given: `$ kubectl get Kind`.

The following will work:
```
$ kubectl get Kind.group.com/resource
$ kubectl get Kind.group.com
```

Reply:
> $ kubectl get Kind.group.com/resource

This string makes no sense.  Try explaining again.  is "resource" actually a "resourcename"?

Reply:
> This string makes no sense. Try explaining again. is "resource" actually a "resourcename"?

Yes, edited my previous comment's example to reflect this better. It should have been `Kind.group.com/resource_name`